src/shogun/mathematics/linalg/GPUBackend.cpp

```diff
@@ -60,9 +61,23 @@ T GPUBackend::dot(const GPUVector<T> &a, const GPUVector<T> &b) const
 #endif
 }
 
+template <typename T>
+T GPUBackend::sum(const GPUVector<T> &vec) const
+{
+#ifdef HAVE_VIENNACL
+	return viennacl::linalg::sum(vec.gpuarray->GPUvec());
+#else
+	SG_SERROR("User did not register GPU backend. \n");
```

Comment:
I don't like it if we would have to put this error message in *every* implementation.

Minor: No space after period.